### PR TITLE
Add note-types to organize finding notes

### PR DESCRIPTION
**Pull request by piyarathnalakmali**
When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR. ([Feature Documentation](https://github.com/DefectDojo/django-DefectDojo/files/3574357/Feature.Documentation.-.DefectDojo.Note.Type.Modification.pdf))
- [x] Model changes must include the necessary migrations in the dojo/dd_migrations folder.
- [x] Add applicable tests to the unit tests.

**The note-type modification has no effect on new or existing installations, or the existing functionalities of the DefectDojo application. All the changes and modifications done by this improvement will be taken into account only if the system admin configures any note-types.**

In the current DefectDojo application there is a functionality for the users to add notes to findings. In this modification a new field “note type” is introduced. For each note added to a finding there should be an associated note-type.

This improvement is useful when it is required to store notes in an organized manner and to restrict closing of findings unless the finding has fulfilled some requirements of notes. 

- As an example, for each finding there is a requirement to know what are the use cases affected by the particular vulnerability. In such case, the admin can add a new note-type as "Use case", thereby allowing the users to add a "Use case" type note to describe the affected use cases.

As part of this feature, a system admin can define note-types according to their requirements. Admin has all the privileges to add, edit, disable and enable note-types as required.

In the previous DefectDojo application, to close a finding user has to add a closing note explaining the reason. With the current improvement, in order to close a finding notes representing all mandatory note-types should also be available. This will ensure that every closed finding has all the details about that finding and its impact on the product. Also any other user can easily verify a closed finding since all notes are properly organized.  

[Feature Documentation - DefectDojo Note Type Modification  .pdf](https://github.com/DefectDojo/django-DefectDojo/files/3574357/Feature.Documentation.-.DefectDojo.Note.Type.Modification.pdf)

**WSO2 uses DefectDojo as the centralized Vulnerability Management System. This modification is a contribution done by WSO2.** WSO2 will use this feature to mandate note types such as: 
- Use case
- Resolution details
- Impact

### CODE REVIEW

File: dojo/db_migrations/0001_initial.py [hunk 1]
```diff
@@ -488,6 +488,17 @@ class Migration(migrations.Migration):
                 'ordering': ['-date'],
             },
         ),
+        migrations.CreateModel(
```
Comment (Maffooch) [inline dojo/db_migrations/0001_initial.py]: Hey @piyarathnalakmali, please do not modify existing migration files, this wrecks existing databases to do this. Instead, create a new one. Check [this](https://docs.djangoproject.com/en/2.2/topics/migrations/) out for more info. Thanks.

**Comment by piyarathnalakmali** [comment]
Hi @Maffooch I'm working on the changes you suggested. Could you please let me know what is the easiest way to run the unit-tests.sh script. I'm getting this error in the travis build and trying to debug the issue.
> FAILED: defectdojo-unit-tests, run `kubectl logs defectdojo-unit-tests --namespace default` for more info
Error: 1 test(s) failed 


Thanks in advance.

**Comment by piyarathnalakmali** [comment]
Hi @Maffooch I did the changes you requested about reporting and migrations. I Will send a separate PR for the documentation to  [dojo docs](https://github.com/DefectDojo/Documentation/tree/master/docs).

Please let me know if there is anything else to be done.

**Comment by devGregA** [comment]
@Maffooch sorry for the delay. Please fix the conflicts & I'll merge.

**Comment by piyarathnalakmali** [comment]
@Maffooch @devGregA thank you for reviewing and merging the PR. Please note that this modification is a contribution done by @WSO2. As mentioned in [sponsorship information](https://github.com/DefectDojo/django-DefectDojo/blob/master/SPONSORING.md) WSO2 would like to receive the points and to be mentioned as a sponsor in future. We are planning to make further contributions to accumulate more points. Documentation PR will follow shortly.

**Comment by devGregA** [comment]
@piyarathnalakmali absolutely. Would you send me the logo / link you would like added to greg.anderson@owasp.org?

**Pull request merged**